### ISSUE TITLE
IllegalArgumentException: Required value was null.

### ISSUE THREAD

**Issue by AliSoftware** [wordpress-mobile/WordPress-Android#15145]
**Sentry Url**: https://sentry.io/share/issue/0685912c5d5640ebbb40f30adc9f69e7/
**User Count**: 60
**Count**: 454
**First Release**: 17.8-rc-1
**First Seen**: 2021-07-16T04:31:22.400000Z
**Last Seen**: 2021-08-06T10:40:24Z
**24 Hours**: 47
**30 Days**: 454

```
IllegalArgumentException: Required value was null.
     at org.wordpress.android.viewmodel.plugins.PluginBrowserViewModel.onSiteChanged(PluginBrowserViewModel.kt:339)
     at org.wordpress.android.fluxc.Dispatcher.emitChange(Dispatcher.java:42)
     at org.wordpress.android.fluxc.store.Store.emitChange(Store.java:32)
     at org.wordpress.android.fluxc.store.SiteStore$onAction$1.invokeSuspend(SiteStore.kt:1155)
...
(7 additional frame(s) were not displayed)
```

**Comment by renanferrari** [comment]
This one seems to be related to https://github.com/wordpress-mobile/WordPress-Android/pull/14812 (cc @planarvoid @develric FYI)

**Comment by develric** [comment]
I was looking a bit into this one. I was not able to reproduce but sharing a few things I was looking into in sentry (and happy to get a double check in case I'm messing up with events and release 😄 ).

The specific error message re this GH issue is below:

```
IllegalArgumentException: Required value was null.
     at org.wordpress.android.viewmodel.plugins.PluginBrowserViewModel.onSiteChanged(PluginBrowserViewModel.kt:339)
     at org.wordpress.android.fluxc.Dispatcher.emitChange(Dispatcher.java:42)
     at org.wordpress.android.fluxc.store.Store.emitChange(Store.java:32)
     at org.wordpress.android.fluxc.store.SiteStore$onAction$1.invokeSuspend(SiteStore.kt:1155)
```

it is triggered by the `requireNotNull` check introduced in [17.7](https://github.com/wordpress-mobile/WordPress-Android/blob/bb9b83e6acc4fcbfdb26ef21edf431b0798487e7/WordPress/src/main/java/org/wordpress/android/viewmodel/plugins/PluginBrowserViewModel.kt#L339)

Actually the issue is that the `SiteStore.getSiteBySiteId` function could return a null value also before it was moved to kotlin (see [SiteStore.java](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/95e3fe545b4c3f8cc4082796c23acbd47e8de008/fluxc/src/main/java/org/wordpress/android/fluxc/store/SiteStore.java#L1530))

Based on the above I would expect that before the 17.7 we should find exceptions like `mSiteStore.getSiteBySiteId(site.siteId) must not be null` when trying to assign null to the not nullable site variable [here](https://github.com/wordpress-mobile/WordPress-Android/blob/3f08c9628b97e039b37c74d43af88dceea9d9748/WordPress/src/main/java/org/wordpress/android/viewmodel/plugins/PluginBrowserViewModel.kt#L339); and we actually find them (using Discover feature):

![image](https://user-images.githubusercontent.com/47797566/133691774-409b16c7-f948-41b1-a482-ce48461a43be.png)

Also filtering the `requireNotNull` related events for the 17.7 (the first release when it was introduced) we get pretty similar low numbers.

![image](https://user-images.githubusercontent.com/47797566/133691933-13e7adf5-58c1-4b7e-849d-ee143f0967c6.png)

Something else seems to have been introduced in 17.8 that made the numbers rise (even though still low)

![image](https://user-images.githubusercontent.com/47797566/133692017-a0b776b1-8273-4553-bb87-89b336fee637.png)

And from there on the numbers stay on similar levels

Here is 17.9
![image](https://user-images.githubusercontent.com/47797566/133692088-11227796-e39e-419f-b582-c30627cdcc66.png)

and 18.0
![image](https://user-images.githubusercontent.com/47797566/133692157-91d47c98-7ab6-4b2d-b6b7-25cf4de736ca.png)

Based on the above I would expect something to have been introduced in 17.8 but the [Release notes](https://github.com/wordpress-mobile/WordPress-Android/blob/6365e6a546ce459a07af5d30fa1381d66308d11a/RELEASE-NOTES.txt#L7) are not helping much AFAIU.

From here I'm not sure how to better proceed; I can think of (not necessarily one excluding the other):

- We invest a bit more time trying to figure out what 17.8 could have introduced that can be related (in the client app or maybe the api? 🤔 ), and we mitigate the possibility (that still could exist sometimes) of having getSiteBySiteId returning a null
- We accept that getSiteBySiteId can return null and we define what's better to happen when this is the case; like not updating the site or, as it happens in the Plugin Activity, we return a error message and [finish the activity](https://github.com/wordpress-mobile/WordPress-Android/blob/ea6da16b74d7c247bfe74a6f74470303c2033008/WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginBrowserActivity.java#L100)
- Other ideas welcome 🙇

**Comment by AliSoftware** [comment]
Moving this issue to 18.5 as I'm doing 18.4 code freeze today.

**Comment by AliSoftware** [comment]
Moving milestone to `18.8` as I'm doing the code freeze for `18.7` now.

Updated numbers for the Sentry issue as of today are: `7.6k events, 1.1k users` so this has raised quite a lot since this was opened. cc @fluiddot to reassess the priority in fixing this.

**Comment by fluiddot** [comment]
I sorted the Sentry issues by the number of events notified and this issue is at 8th position, in relation to the latest released version `18.5` here are some metrics:
- Total users affected by the issue: **1.106**
- Total users in `18.5` (Nov 15): **237.884**
- Total users in `18.5` version affected: **106**
- Rate of users impacted by this issue in `18.5`: **0,04%**

Taking the priority matrix, I understand this issue is a high severity and low impact since the rate of impacted users is below 5%, so we could keep the Medium priority.

**Comment by AliSoftware** [comment]
Updated Sentry numbers as of today:
 - Events: 8562
 - Users affected: 1110

Seems not much more users than the last report, so still steady but at least it's not skyrocketing much since 14 days ago.
![image](https://user-images.githubusercontent.com/216089/143879858-5b583894-e277-4e40-9653-ffb6e2decf6f.png)

Moving milestone to 18.9 as I'm code-freezing 18.8 right now.
